File: app/src/main/java/com/bigscreen/mangindo/stored/PreferenceService.java
package com.bigscreen.mangindo.stored;


import android.content.Context;
import android.content.SharedPreferences;

public class PreferenceService {

    private SharedPreferences sharedPreference;

    public PreferenceService(Context context) {
        sharedPreference = context.getSharedPreferences(context.getPackageName(), Context.MODE_PRIVATE);
    }

    public String getString(String key, String defValue) {
        return sharedPreference.getString(key, defValue);
    }

    public void saveString(String key, String value) {
        sharedPreference.edit().putString(key, value).apply();
    }

    public int getInt(String key, int defValue) {
        return sharedPreference.getInt(key, defValue);
    }

    public void saveInt(String key, int value) {
        sharedPreference.edit().putInt(key, value).apply();
    }

    public boolean getBoolean(String key, boolean defValue) {
        return sharedPreference.getBoolean(key, defValue);
    }

    public void saveBoolean(String key, boolean value) {
        sharedPreference.edit().putBoolean(key, value).apply();
    }

    public long getLong(String key, long defValue) {
        return sharedPreference.getLong(key, defValue);
    }

    public void saveLong(String key, long value) {
        sharedPreference.edit().putLong(key, value).apply();
    }

    public void remove(String key) {
        sharedPreference.edit().remove(key).apply();
    }

    public void clear() {
        sharedPreference.edit().clear().apply();
    }
}

File: app/src/main/java/com/bigscreen/mangindo/stored/PreferenceService.kt
package com.bigscreen.mangindo.stored

import android.content.Context
import android.content.Context.MODE_PRIVATE

class PreferenceService(context: Context) {

    private val sharedPreference by lazy {
        context.getSharedPreferences(context.packageName, MODE_PRIVATE)
    }

    fun getString(key: String, defValue: String): String? {
        return sharedPreference.getString(key, defValue)
    }

    fun saveString(key: String, value: String) {
        sharedPreference.edit().putString(key, value).apply()
    }

    fun getInt(key: String, defValue: Int): Int {
        return sharedPreference.getInt(key, defValue)
    }

    fun saveInt(key: String, value: Int) {
        sharedPreference.edit().putInt(key, value).apply()
    }

    fun getBoolean(key: String, defValue: Boolean): Boolean {
        return sharedPreference.getBoolean(key, defValue)
    }

    fun saveBoolean(key: String, value: Boolean) {
        sharedPreference.edit().putBoolean(key, value).apply()
    }

    fun getLong(key: String, defValue: Long): Long {
        return sharedPreference.getLong(key, defValue)
    }

    fun saveLong(key: String, value: Long) {
        sharedPreference.edit().putLong(key, value).apply()
    }

    fun remove(key: String) {
        sharedPreference.edit().remove(key).apply()
    }

    fun clear() {
        sharedPreference.edit().clear().apply()
    }
}